compiler/src/dotty/tools/dotc/transform/Splicer.scala

```diff
@@ -141,7 +141,7 @@ object Splicer {
       loadModule(fn.moduleClass)
 
     protected def interpretNew(fn: Symbol, args: => List[Result])(implicit env: Env): Object = {
-      val clazz = loadClass(fn.owner.fullName)
+      val clazz = loadClass(fn.owner.fullName.toString)
```

Comment:
What if the class is an inner class?

Reply:
Inner classes are not supported and this method is never called for them.